src/main.lua

```diff
@@ -1638,43 +1642,32 @@ addEventingHandler("distribute", utils.hasMatchingTag("Action", "Tick"), functio
 		local tickTimestamp = math.min(msgTimestamp or 0, epochDistributionTimestamp)
 		-- TODO: if we need to "recover" epochs, we can't rely on just the current message hashchain and block height,
 		-- we should set the prescribed observers and names to empty arrays and distribute rewards accordingly
-		local tickSucceeded, resultOrError = pcall(tick.tickEpoch, tickTimestamp, blockHeight, hashchain, msgId)
-		if tickSucceeded then
-			if tickTimestamp == epochDistributionTimestamp then
-				-- if we are distributing rewards, we should update the last ticked epoch index to the current epoch index
-				LastTickedEpochIndex = i
-				table.insert(tickedEpochIndexes, i)
-			end
-			ao.send({
-				Target = msg.From,
-				Action = "Tick-Notice",
-				LastTickedEpochIndex = LastTickedEpochIndex,
-				Data = json.encode(resultOrError),
-			})
-			if resultOrError.maybeNewEpoch ~= nil then
-				table.insert(newEpochIndexes, resultOrError.maybeEpoch.epochIndex)
-			end
-			if resultOrError.maybeDemandFactor ~= nil then
-				table.insert(newDemandFactors, resultOrError.maybeDemandFactor)
-			end
-			if resultOrError.pruneGatewaysResult ~= nil then
-				table.insert(newPruneGatewaysResults, resultOrError.pruneGatewaysResult)
-			end
-			if resultOrError.maybeDistributedEpoch ~= nil then
-				tickedRewardDistributions[tostring(resultOrError.maybeDistributedEpoch.epochIndex)] =
-					resultOrError.maybeDistributedEpoch.distributions.totalDistributedRewards
-				totalTickedRewardsDistributed = totalTickedRewardsDistributed
-					+ resultOrError.maybeDistributedEpoch.distributions.totalDistributedRewards
-			end
-		else
-			-- reset the state to previous state
-			ao.send({
-				Target = msg.From,
-				Action = "Invalid-Tick-Notice",
-				Error = "Invalid-Tick",
-				Data = json.encode(resultOrError),
-			})
-			-- TODO: But keep ticking ahead?!? We just need be to robust against potential failures here
+		local resultOrError = tick.tickEpoch(tickTimestamp, blockHeight, hashchain, msgId)
```

Comment:
will change to tickResult

Reply:
[471c1da](https://github.com/ar-io/ar-io-network-process/pull/210/commits/471c1da1aae9bcdf868bec12bd6caa2c610d57a5)